lua/conform/formatters/deno_fmt.lua

```diff
@@ -16,11 +34,29 @@ return {
   },
   command = "deno",
   args = function(self, ctx)
-    return {
+    local extension = extensions[vim.bo[ctx.buf].filetype]
+    local formatter_args = {
       "fmt",
       "-",
       "--ext",
-      extensions[vim.bo[ctx.buf].filetype],
+      extension,
     }
+
+    if
+      vim.tbl_get(unstable_extensions, extension)
+      and not vim.list_contains(
+        vim.tbl_get(conform.formatters, "deno_fmt", "append_args") or {},
+        "--unstable-component"
+      )
```

Comment:
I don't love this reaching in to `conform.formatters` to check `append_args`. Will the command error if this argument is supplied more than once? Also, if the user needs fine-grained control over the arguments they can always override the `args` function themselves. Lastly, if we absolutely must, we could add another option to this formatter that allows the user to disable the auto-adding of this arg (but only if the first two options are insufficient for some reason).

Reply:
> Will the command error if this argument is supplied more than once?

Unfortunately, yes. The formatter exists with an error saying that the same flag should only be used once:
![image](https://github.com/user-attachments/assets/7f4a02e6-8c4d-4e09-88cc-d7f9a9463683)

But I agree; it feels rather clunky to reach into `append_args`. I think we could blatantly add `--unstable-component` since the error message indicates that this plugin already has the flag if you try to configure it yourself.

For context, we could skip the `if unstable_extensions[extension]` check since the `deno fmt` works fine if you provide the `--unstable-component` flag for _stable_ files (JavaScript, TypeScript, etc.), but I think it's better to add it only where needed (since it's unstable in some form).

> Also, if the user needs fine-grained control over the arguments they can always override the args function themselves. Lastly, if we absolutely must, we could add another option to this formatter that allows the user to disable the auto-adding of this arg (but only if the first two options are insufficient for some reason).

Good idea! 👍